framework/components/ecl-lang-select-pages/_lang-select-pages.scss

```diff
@@ -1,142 +1,105 @@
 /*
-Language select (Page)
-*/
+ * Language selector (page level)
+ * @define lang-select-page
+ */
 
-.lang-select-page {
+.ecl-lang-select-page {
   @include ecl-hidden-print();
 
-  width: 100%;
-  overflow: hidden;
+  align-items: center;
   background-color: map-get($ecl-colors, 'grey-5');
-  font-size: map-get($ecl-font-size, 'xs');
   border-bottom: 1px solid map-get($ecl-colors, 'grey-50');
+  color: $ecl-color-shade;
+  display: flex;
+  flex-wrap: wrap;
+  font-size: map-get($ecl-font-size, 'xs');
+  margin: 0;
 }
 
-.ecl-icon.lang-select-page__icon {
+.ecl-lang-select-page__icon {
   @include ecl-hidden-print();
 
-  float: left;
-  margin: 0;
-  padding: 0;
-  font-size: 2em;
+  flex-grow: 0;
+  flex-shrink: 0;
+  font-size: map-get($ecl-font-size, 'xl');
+}
+
+.ecl-lang-select-page__unavailable {
+  padding: map-get($ecl-spacing, 's') map-get($ecl-spacing, 'xs');
+  text-decoration: line-through;
 
-  &::before {
-    float: left;
-    padding-top: 30%;
+  @include ecl-media-breakpoint-up(sm) {
+    margin-right: map-get($ecl-spacing, 'l');
   }
 }
 
-.lang-select-page__unavailable {
-  text-decoration: line-through;
+.ecl-lang-select-page__list {
+  align-items: stretch;
+  align-self: stretch;
+  display: flex;
+  flex-grow: 1;
+  flex-wrap: wrap;
+  margin: 0;
+  padding: 0;
 }
 
-ul.lang-select-page__list {
-  float: left;
-  margin-bottom: 0 !important;
-  padding-left: 0 !important;
-
-  .lang-select-page__option {
-    list-style: none;
-    float: left;
-    padding: 0.85em;
-
-    a {
-      display: block;
-    }
-
-    &.is-selected {
-      background-color: map-get($ecl-colors, 'yellow-100');
-      font-weight: bold;
-      color: #000;
-    }
-  }
+.ecl-lang-select-page__option {
+  align-items: center;
+  display: flex;
+  list-style: none;
+  padding: map-get($ecl-spacing, 's') map-get($ecl-spacing, 'xs');
+}
+
+.ecl-lang-select-page__option--is-selected {
+  background-color: map-get($ecl-colors, 'yellow-100');
+  color: #000;
+  font-weight: bold;
 }
 
-select.lang-select-page__list {
+.ecl-lang-select-page__dropdown {
   @include ecl-hidden-print();
 
-  width: 10em;
-  height: 3.4em;
-  padding-left: 1.2em;
-  line-height: 3.4em;
-  background: transparent url("../images/png/arrow-thick-down.png") no-repeat 8em center;
-  font-weight: bold;
+  align-self: stretch;
+  appearance: none;
+  background: transparent url('../images/png/arrow-thick-down.png') no-repeat right map-get($ecl-spacing, 'xs') center;
```

Comment:
It would be much better indeed. Do we have the SVG equivalent? (I'll have a look)